src/engine/ReactPlugin/ToolReactPlugin.js

```diff
@@ -32,13 +32,9 @@ export function withToolPlugin(ReactComponent, methods = []) {
      */
     useEffect(() => {
       PluginManagerIDE.resetBookmarks();
-      on(
-        PLUGINS.TABS.NAME,
-        PLUGINS.TABS.ON.ACTIVE_TAB_CHANGE,
-        async ({ id }) => {
-          if (profile.name === id) PluginManagerIDE.resetBookmarks();
-        }
-      );
+      on(PLUGINS.TABS.NAME, PLUGINS.TABS.ON.ACTIVE_TAB_CHANGE, async () => {
+        PluginManagerIDE.resetBookmarks();
```

Comment:
Instead of just resetting the bookmarks when the profile.name and id of the new Active Tab are the same (don't forget this code will run for all the opened tabs as this is a decorator), we are now resetting the bookmarks for all tabs on tab change. With the new way of opening multiple tools, we now have different id's for the tools, which means we can't compare them with the `profile.name`, unless we do some workarounds.

Reply:
That is a bit odd, could we instead make the comparison differently?